src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -112,6 +113,15 @@ public MainTable(MainTableDataModel model, JabRefFrame frame,
         // Enable sorting
         model.getEntriesFilteredAndSorted().comparatorProperty().bind(this.comparatorProperty());
 
+        model.getEntriesFilteredAndSorted().addListener(
```

Comment:
Although it is a neat idea to use the listener, I'm afraid that it does not work as intended. The listener is also triggered if I clear a search as then a bunch of previously filtered-out entries are readded to the list. What should work is if you add a listener to `database.getDatabase()`.

Reply:
I ran into multiple race conditions after I added a change listener to the internal observable list of bib entries in the database. Therefore I choose to introduce a new event that is sent over the event bus when the insertion of one or many entries is finished.